File: search/single_repo.py
import json
import os
import re


def enum_filename(repo_dir_path,inode_name,isFile=True):
    indoes = []    
    if '*' in inode_name:
        inode_name = inode_name.replace('*','')
    for file in os.listdir(repo_dir_path):
        if isFile:
            if os.path.isdir(os.path.join(repo_dir_path,file)):
                indoes.extend(enum_filename(os.path.join(repo_dir_path,file),inode_name))
            elif file.endswith(inode_name):
                indoes.append(file)
        else:
            if os.path.isdir(os.path.join(repo_dir_path,file)):
                # TODO Recurse through directories
                if file.endswith(inode_name):
                    indoes.append(file)
    return indoes

def enumerate_code_blocks(readme,setup):
    code_blocks_start_end = readme.split('```')
    code_blocks = []
    for i in range(len(code_blocks_start_end)):
        if i % 2 == 1:
            code_blocks.append(code_blocks_start_end[i])
    
    if len(code_blocks) > 0:
        setup.setup = code_blocks[0]
        for code_block in code_blocks:
            if 'pip install' in code_block:
                install = 'pip install ' +code_block.split('pip install')[1].strip()
                if 'requirements.txt' in install:
                    setup.clone = True
                setup.installs.append(install)
        
        with open('code_blocks.json', 'w') as f:
            json.dump(code_blocks, f)
        

def enum_python_files(repo_dir_path):
    python_files = []
    for file in os.listdir(repo_dir_path):
        if os.path.isdir(os.path.join(repo_dir_path,file)):
            python_files.extend(enum_python_files(os.path.join(repo_dir_path,file)))
        elif file.endswith('.py'):
            python_files.append(file)
    return python_files

File: src/search/single_repo.ts
import * as fs from 'fs';
import { join } from 'path';


export function enumFilename(repoDirPath: string, inodeName: string, isFile: boolean = true): string[] {
    const indoes: string[] = [];

    if (inodeName.includes('*')) {
        inodeName = inodeName.replace('*', '');
    }

    for (const file of fs.readdirSync(repoDirPath)) {
        if (isFile) {
            if (fs.lstatSync(join(repoDirPath, file)).isDirectory()) {
                indoes.push(...enumFilename(join(repoDirPath, file), inodeName));
            } else if (file.endsWith(inodeName)) {
                indoes.push(file);
            }
        } else {
            if (fs.lstatSync(join(repoDirPath, file)).isDirectory()) {
                // TODO Recurse through directories
                if (file.endsWith(inodeName)) {
                    indoes.push(file);
                }
            }
        }
    }

    return indoes;
}

export function enumerateCodeBlocks(readme: string, setup: any): void {
    const codeBlocksStartEnd: string[] = readme.split('```');
    const codeBlocks: string[] = [];

    for (let i = 0; i < codeBlocksStartEnd.length; i++) {
        if (i % 2 === 1) {
            codeBlocks.push(codeBlocksStartEnd[i]);
        }
    }

    if (codeBlocks.length > 0) {
        setup.setup = codeBlocks[0];

        for (const codeBlock of codeBlocks) {
            if (codeBlock.includes('pip install')) {
                const install: string =
                    'pip install ' + codeBlock.split('pip install')[1].trim();

                if (install.includes('requirements.txt')) {
                    setup.clone = true;
                }

                setup.installs.push(install);
            }
        }

        const codeBlocksJson: string = JSON.stringify(codeBlocks);

        fs.writeFileSync('code_blocks.json', codeBlocksJson);
    }
}

export function enumPythonFiles(repoDirPath: string): string[] {
    const pythonFiles: string[] = [];

    for (const file of fs.readdirSync(repoDirPath)) {
        if (fs.lstatSync(join(repoDirPath, file)).isDirectory()) {
            pythonFiles.push(...enumPythonFiles(join(repoDirPath, file)));
        } else if (file.endsWith('.py')) {
            pythonFiles.push(file);
        }
    }

    return pythonFiles;
}